src/core/createFrameResponse.ts

```diff
@@ -0,0 +1,58 @@
+import { NextResponse } from 'next/server';
+import { Button, FrameMetadata } from './sharedTypes';
+
+/**
+ * Creates a response with a new valid frame to be rendered.
+ *
+ * @param buttons: The buttons to use for the frame.
+ * @param image: The image to use for the frame.
+ * @param post_url: The URL to post the frame to.
+ * @param refresh_period: The refresh period for the image used.
+ * @returns An HTML response containing metadata for the frame.
+ */
+function createFrameResponse({
+  buttons,
+  image,
+  post_url,
+  refresh_period,
+}: FrameMetadata): NextResponse {
+  const imageHtml = image ? `<meta property="fc:frame:image" content="${image}" />` : '';
+  // Ensure only up to 4 buttons are processed
+  let buttonsHtml = '';
+  if (buttons) {
+    const validButtons = buttons
+      .slice(0, 4)
+      .filter((button): button is Button => button !== undefined);
```

Comment:
I think also those checks are not need it, and `FrameMetadata` type should cover those automatically.

Reply:
Removed these checks.